### DISCUSSION TITLE
Circumvent package size limit

### DISCUSSION THREAD

**Discussion by vincentarelbundock** [r-universe-org/help#454] [General]
Hi,

Thanks for your essential work on R-Universe!

My `marginaleffects` package repository includes largish data files which are required for continuous integration testing on Github Actions. When R-Universe tries to build the Windows binary, [the process errors because the package exceeds the default 100M size limit (error pasted below).](https://github.com/r-universe/vincentarelbundock/actions/runs/10169837947/job/28128115688)

I know that CRAN imposes a similar limit. When I submit packages there, I manually add large files to .Rbuildignore.

Is it possible to do something like that on R-Universe? Can I customize the build actions to either allow bigger packages or ignore some files? How do you recommend I circumvent the file size limit?

Thanks!

```sh
Run if [ -n "$(find "marginaleffects_0.21.0.14.zip" -prune -size +100M)" ]; then
  if [ -n "$(find "marginaleffects_0.21.0.14.zip" -prune -size +100M)" ]; then
    echo "File marginaleffects_0.21.0.14.zip is larger than 100 MB. This is not allowed"
    exit 1
  fi
  shell: C:\Program Files\Git\bin\bash.EXE --noprofile --norc -e -o pipefail {0}
  env:
    UNIVERSE_NAME: vincentarelbundock
    MY_UNIVERSE: https://vincentarelbundock.r-universe.dev/
    GH_APP_ID: 87942
    RTOOLS42_HOME: C:\rtools44
    RTOOLS43_HOME: C:\rtools44
    _R_INSTALL_TIME_PATCHES_: https://www.r-project.org/nosvn/winutf8/ucrt3/
    JAVA_HOME: C:\hostedtoolcache\windows\Java_Temurin-Hotspot_jdk\17.0.12-7\x64
File marginaleffects_0.21.0.14.zip is larger than 100 MB. This is not allowed
Error: Process completed with exit code 1.
```

**Comment by jeroen** [comment]
I think the same way as CRAN? If you only need those files for CI, add them to `.Rbuildignore` such that they do not get included in the deployed packages on r-universe.

**Reply by vincentarelbundock** [reply]
Thanks. The problem is that if I add them to `.Rbuildignore` in the main repository, my CI test suite will no longer run on Github Actions.

**Reply by jeroen** [reply]
Perhaps you can add them to `.Rbuildignore` and add a step to your own CI workflow to remove this line? Or you can condition something on the presence of the `CI` environment variable, which is set in GHA but not on CRAN/R-universe?

**Reply by vincentarelbundock** [reply]
lol, I feel ridiculous for not having thought of this obvious solution. Thanks a lot for your time and suggestion. And thanks again for your work on R-universe!
